methods.py

```diff
@@ -828,7 +828,10 @@ def get_compiler_version(env):
                     sem_ver = split[1].split(".")
                     ret["major"] = int(sem_ver[0])
                     ret["minor"] = int(sem_ver[1])
-                    ret["patch"] = int(sem_ver[2])
+                    if ("Preview" in sem_ver[2]):
+                        ret["patch"] = int(sem_ver[2].split(" ")[0])
+                    else:
+                        ret["patch"] = int(sem_ver[2])
```

Comment:
Because `split()` just returns the original string as the first index if no splitting occurs & whitespace is the implicit separator, this can be simplified to a one-liner:
```suggestion
                    ret["patch"] = int(sem_ver[2].split()[0])
```